packages/angular/src/lib/get-variant.pipe.ts

```diff
@@ -0,0 +1,30 @@
+import { Pipe, PipeTransform } from "@angular/core";
+import { ABConfig } from "@tryabby/core";
+import { Observable } from "rxjs";
+import { Key } from "ts-toolbelt/out/Any/Key";
+import { AbbyService, ExtractVariants } from "./abby.service";
+
+@Pipe({
+  name: "getAbbyVariant",
+})
+export class GetAbbyVariantPipe<TestName extends Key, Tests extends Record<TestName, ABConfig>>
+  implements PipeTransform
+{
+  constructor(private abbyService: AbbyService<string, string, Tests>) {}
+
+  transform(testName: TestName): Observable<string>;
+  transform<S>(
+    testName: TestName,
+    lookupObject: Record<ExtractVariants<TestName, Tests>, S>
+  ): Observable<S>;
+  transform<S>(
+    testName: TestName,
+    lookupObject?: Record<ExtractVariants<TestName, Tests>, S>
+  ): Observable<string | S> {
+    if (lookupObject === undefined) {
+      return this.abbyService.getVariant(testName);
+    }
+
+    return this.abbyService.getVariant(testName, lookupObject);
```

Comment:
```suggestion
    return this.abbyService.getVariant(testName, lookupObject);
```

Reply:
I also wanted to do it like this, but the parameter is an optional one, so existing code doesn't break.
You can either not add the parameter or pass it, but then it is not allowed to be `undefined`:
![image](https://github.com/tryabby/abby/assets/29003122/2e7fbe8e-2f4c-4b30-8fd1-87641c56f091)
The only option would be to add `| undefined` to the parameter like this:
```ts
lookupObject?: { [key in ExtractVariants<T, Tests>]: S } | undefined
```
But it kinda looks ugly